### PR TITLE
Branchs in TreeView now collapse only if it was selected first

### LINKED ISSUE #18: Collapse group in explorer only when already selected
### Current behaviour

1. Click on `entry_0000` group => group node expands and group metadata appears in metadata viewer.
2. Click on `start_time` dataset => dataset metadata appears in metadata viewer.
3. Click on `entry_0000` group again to see its metadata => **group node collapses** and group metadata appears in metadata viewer.

### Suggested behaviour

3. Click on `entry_0000` group again to see its metadata => group metadata appears in metadata viewer (group node does _not_ collapse).
4. Click on `entry_0000` group one more time => group node collapses.

### PR DESCRIPTION
**Pull request by loichuder**
Fix #18

### CODE REVIEW

File: src/h5web/explorer/TreeView.tsx [hunk 1]
```diff
@@ -39,6 +39,9 @@ function TreeView<T>(props: Props<T>): JSX.Element {
                 onSelect(node);
 
                 if (isBranch) {
+                  if (isExpanded && node !== selectedNode) {
+                    return; // Disable collapsing if the node is not selected
+                  }
```
Comment (axelboc-esrf) [inline src/h5web/explorer/TreeView.tsx]: Tu peux tout mettre dans la même condition, non ?

Comment (loichuder) [reply src/h5web/explorer/TreeView.tsx]: I can but it is ugly: `isBranch && (!isExpanded || node === selectedNode)`

Comment (loichuder) [reply src/h5web/explorer/TreeView.tsx]: Perhaps better now: `isBranch && (!isExpanded || isSelected)`

Comment (axelboc-esrf) [reply src/h5web/explorer/TreeView.tsx]: yes ! 💯

**Pull request merged**